### ISSUE TITLE
Craft4 Beta4, Imager-X Beta, 100% transparent webp?

### ISSUE THREAD

**Issue by MattWilcox** [spacecatninja/craft-imager-x#143]
### I'm submitting a... 
- [ x ] bug report
- [ ] feature request
- [ ] general question

### Description
All transforms that are outputting webp are creating images of the correct dimension, but with no content - they're 100% transparent. Things are working fine with jpgs. The server seems OK for webp file generation as the native Craft transforms are generating proper webp images in this project. I've not been able to see any mis-configuration in how I've got this set up.

imager-x.php
```
<?php
// General ImagerX settings here, see:
// https://imager-x.spacecat.ninja/configuration.html#general-configuration

use craft\helpers\App;

return [
    '*' => [
        'interlace' => true,
        'allowUpscale' => true,
        'safeFileFormats' => ['jpg', 'jpeg', 'gif', 'png', 'webp']
    ],

    'dev' => [
        // Nitro has webp support in ImageMagik and so does not need Custom Encoders to handle them
        // 'mockImage' => '/assets/images/missing-image.png',
        'cacheEnabled' => false,
        'suppressExceptions' => false
    ],

    'staging' => [
        // Ubuntu doesn't have webp support baked into ImageMagik, so we provide cwebp
        'customEncoders' => [
            'webp' => [
                'path' => App::env('PATH_TO_CWEBP'),
                'options' => [
                    'quality' => 80,
                    'effort' => 4,
                ],
                'paramsString' => '-q {quality} -m {effort} {src} -o {dest}'
            ],
        ],
    ],

    ...
];
```

imager-x-transforms.php
```
<?php
// Named Transforms are stored here, see:
// https://imager-x.spacecat.ninja/usage/named-transforms.html#nested-named-transforms

return [
    'bannerImageJpg' => [
        'defaults' => [
            'format' => 'jpg',
            'mode' => 'crop',
            'ratio' => 2.39/1,
            'jpegQuality' => 80
        ],
        'transforms' => [
            ['width' => 480, 'ratio' => '16/9'],
            ['width' => 960],
            ['width' => 1500],
            ['width' => 2200]
        ]
    ],
    'bannerImageWebp' => [
        'defaults' => [
            'format' => 'webp',
            'webpQuality' => 80
        ],
        'transforms' => 'bannerImageJpg'
    ]
];
```

imager-x-generate.php
```
<?php
// when to generate which Named Transforms, see:
// https://imager-x.spacecat.ninja/usage/generate.html#setting-it-up

return [
    'elements' => [
        [
            'elementType' => \craft\elements\Entry::class,
            'criteria' => [
                'section' => 'pages'
            ],
            'fields' => [
                'listingImage'
            ],
            'transforms' => [
                ['bannerImageJpg', 'bannerImageWebp']
            ]
        ]
    ]
];
```

page.twig
```
{% if entry.listingImage | length %}
	{% set jpgs  = craft.imagerx.transformImage( entry.listingImage.one(), 'bannerImageJpg') %}
	{% set webps = craft.imagerx.transformImage( entry.listingImage.one(), 'bannerImageWebp') %}

	<ul>
		<li>Browser Support for WEBP: {{ craft.imager.clientSupports('webp') ? 'yes' : 'no' }}</li>
		<li>Browser Support for AVIF: {{ craft.imager.clientSupports('avif') ? 'yes' : 'no' }}</li>
		<li>Browser Support for JPEG XL: {{ craft.imager.clientSupports('jxl') ? 'yes' : 'no' }}</li>
		<li>Server Support for WEBP: {{ craft.imager.serverSupportsWebp() ? 'yes' : 'no' }}</li>
		<li>Server Support for AVIF: {{ craft.imager.serverSupportsAvif() ? 'yes' : 'no' }}</li>
		<li>Server Support for JPEG XL: {{ craft.imager.serverSupportsJxl() ? 'yes' : 'no' }}</li>
	</ul>

	<style>picture > img { width: 100%; height: auto; }</style>
	<picture>
		<!-- prefer webp, list these first. Order matters, first media to match is what's rendered - opposite of CSS rules -->
		<source media="(max-width: 359px)" srcset="{{ webps[0].url }}" type="{{ webps[0].mimeType }}" width="{{ webps[0].width }}" height="{{ webps[0].height }}">
		<source media="(min-width: 680px)" srcset="{{ webps[2].url }}" type="{{ webps[2].mimeType }}" width="{{ webps[2].width }}" height="{{ webps[2].height }}">
		<source media="(min-width: 360px)" srcset="{{ webps[1].url }}" type="{{ webps[1].mimeType }}" width="{{ webps[1].width }}" height="{{ webps[1].height }}">

		<!-- jpg fallbacks -->
		<source media="(max-width: 359px)" srcset="{{ jpgs[0].url }}" type="{{ jpgs[0].mimeType }}" width="{{ jpgs[0].width }}" height="{{ jpgs[0].height }}">
		<source media="(min-width: 680px)" srcset="{{ jpgs[2].url }}" type="{{ jpgs[2].mimeType }}" width="{{ jpgs[2].width }}" height="{{ jpgs[2].height }}">
		<source media="(min-width: 360px)" srcset="{{ jpgs[1].url }}" type="{{ jpgs[1].mimeType }}" width="{{ jpgs[1].width }}" height="{{ jpgs[1].height }}">

		<img
			loading="lazy"
			decoding="async"
			src="{{ jpgs[0].url }}"
			width="{{ jpgs[0].width }}"
			height="{{ jpgs[0].height }}"
			alt="{{ entry.listingImage.one().alt }}"
		/>
	</picture>
{% endif %}
```

<img width="979" alt="Screenshot 2022-04-26 at 16 28 41" src="https://user-images.githubusercontent.com/814039/165336865-de038e3a-9347-4d28-b550-276d630dacd9.png">

And then if I just remove the webp `source` lines from the `picture`...

<img width="979" alt="Screenshot 2022-04-26 at 16 30 05" src="https://user-images.githubusercontent.com/814039/165337258-c0c78576-4fa9-4496-b819-85c259ddb875.png">

---
- Imager version & edition: 4.0.0.beta.3
- Imager transformer: craft
- Craft version: 4.0.0-beta.3
- PHP version: 8.0
- Nitro2

**Comment by aelvan** [comment]
Hi, thanks for letting me know. I'll test and get back to you!

**Comment by aelvan** [comment]
This should be fixed in 4.0.0-beta.4.
